rj_param_utils/src/ros2_param_provider.cpp

```diff
@@ -68,55 +121,7 @@ void ROS2ParamProvider::InitUpdateParamCallbacks(rclcpp::Node* node) {
     using rcl_interfaces::msg::SetParametersResult;
     const auto on_update =
         [this](const std::vector<rclcpp::Parameter>& params) -> SetParametersResult {
-        bool all_succeeded = true;
-        for (const rclcpp::Parameter& param : params) {
-            std::string full_name = ConvertFullNameFromROS2(param.get_name());
-            bool success = false;
-            switch (param.get_type()) {
-                case rclcpp::PARAMETER_BOOL:
-                    success = TryUpdate(full_name, param.as_bool());
-                    break;
-                case rclcpp::PARAMETER_INTEGER:
-                    success = TryUpdate(full_name, param.as_int());
-                    break;
-                case rclcpp::PARAMETER_DOUBLE:
-                    success = TryUpdate(full_name, param.as_double());
-                    break;
-                case rclcpp::PARAMETER_STRING:
-                    success = TryUpdate(full_name, param.as_string());
-                    break;
-                case rclcpp::PARAMETER_BYTE_ARRAY:
-                    success = TryUpdate(full_name, param.as_byte_array());
-                    break;
-                case rclcpp::PARAMETER_BOOL_ARRAY:
-                    success = TryUpdate(full_name, param.as_bool_array());
-                    break;
-                case rclcpp::PARAMETER_INTEGER_ARRAY:
-                    success = TryUpdate(full_name, param.as_integer_array());
-                    break;
-                case rclcpp::PARAMETER_DOUBLE_ARRAY:
-                    success = TryUpdate(full_name, param.as_double_array());
-                    break;
-                case rclcpp::PARAMETER_STRING_ARRAY:
-                    success = TryUpdate(full_name, param.as_string_array());
-                    break;
-                default:
-                    break;
-            }
-
-            // Failure could happen in two main cases:
-            //  - The parameter does not exist
-            //  - The parameter's type as declared does not match the parameter's current type
-            if (!success) {
-                // NOLINTNEXTLINE(bugprone-lambda-function-name)
-                SPDLOG_WARN("Failed to set parameter {}", param.get_name());
-                all_succeeded = false;
-            }
-        }
-
-        SetParametersResult set_parameters_result;
-        set_parameters_result.successful = all_succeeded;
-        return set_parameters_result;
+        UpdateParameters(params);
```

Comment:
This lambda needs to return something.
```suggestion
        return UpdateParameters(params);
```
Also, then you shouldn't have to explicitly specify that it returns `SetParametersResult` above.